app/src/main/java/hr/foi/air/servicesync/backend/FirestoreCompanyDetails.kt

```diff
@@ -1,50 +1,104 @@
 package hr.foi.air.servicesync.backend
 
-import android.content.Context
+import android.util.Log
+import com.google.firebase.firestore.FirebaseFirestore
 import com.google.firebase.firestore.GeoPoint
-import hr.foi.air.servicesync.data.CompanyInstance
 
 class FirestoreCompanyDetails {
 
-    private var companyName: String = "No name found!"
+    private val db = FirebaseFirestore.getInstance()
+
     private var companyDescription: String = "No description found!"
     private var companyCategory: String = "No category found!"
     private var companyWorkingHours: Int = 0
     private var companyGeoPoint: GeoPoint = GeoPoint(0.0, 0.0)
 
-    fun loadCompanyName(context: Context, onResult: (String?) -> Unit) {
-        CompanyInstance.fetchCompanyName(context) { name ->
-            companyName = name ?: "No name found!"
-            onResult(companyName)
-        }
+    fun loadCompanyDescriptionByName(companyName: String, onResult: (String?) -> Unit) {
```

Comment:
You have implemented reusability for this class that used to be only for mock data. Thanks for listening the team :-)